From 48be27e81edd59cd323b5fe1f60ac3062868cdb5 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 17 Jul 2012 18:06:35 +0800
Subject: [PATCH 329/449] ENGR00217427 VPU unit test bug fix: VPU output has
 mosaics when frame buffer is in tiled frame format
 and is allocated by VPU

Bug:
- VPU output has mosaics when frame buffer is in tiled frame format and
  is allocated by VPU.
- If frame buffer is in Linear format or tiled field format, correct
- If frame buffer is allocated by IPU, correct

Root cause:
In tiled frame format, mvColBuf is calculated based on chr_bot_base
which is 0

Fix:
Calculate mvColBuf based on chr_top_base

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/fb.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/test/mxc_vpu_test/fb.c b/test/mxc_vpu_test/fb.c
index 0f96fa2..bf74c94 100644
--- a/test/mxc_vpu_test/fb.c
+++ b/test/mxc_vpu_test/fb.c
@@ -252,8 +252,13 @@ struct frame_buf *tiled_framebuf_alloc(int stdMode, int format, int strideY, int
 	fb->addrCr = chr_bot_20bits << 16;
 	fb->strideY = strideY;
 	fb->strideC = strideY / divX;
-	if (mvCol)
-		fb->mvColBuf = chr_bot_base + chroma_bot_size;
+	if (mvCol) {
+		if (mapType == TILED_FRAME_MB_RASTER_MAP) {
+			fb->mvColBuf = chr_top_base + chroma_top_size;
+		} else {
+			fb->mvColBuf = chr_bot_base + chroma_bot_size;
+		}
+	}
 
     return fb;
 }
-- 
1.8.0

